internal/cmd/addhooks/prepush.go

```diff
@@ -0,0 +1,22 @@
+package addhooks
+
+var prePushScript = []byte(`#!/bin/sh
+set -e
+
+remote="$1"
+url="$2"
+
+if ! command -v gittuf > /dev/null
+then
+    echo "gittuf could not be found"
+    echo "Download from: https://github.com/gittuf/gittuf/releases/latest"
+    exit 1
+fi
+
+echo "Pulling RSL from ${remote}."
+gittuf rsl remote pull ${remote}
```

Comment:
We have a slight issue with branching here where we don't check if rsl update includes changes to the branch currently being pushed. I don't think we can do that strictly within the hook, maybe using a more sophisticated gittuf command.

Since the HEAD branch is not updated but the RSL is, we can't run `verify-ref` after the pull either, because that'll fail simply because the HEAD != expected value in the RSL.

Reply:
I see the problem you are describing, but currently we do not run any verification in the script. Is this something we should do?

Reply:
It's fine for now, I was also thinking aloud here.